### PR TITLE
change substr() check with strrpos at InjectTemplateListener::deriveControllerClass()

### PR DESCRIPTION
**Pull request by samsonasik**
- [x] Is this related to quality assurance?

### CODE REVIEW

**Comment by samsonasik** [comment]
after some check, it is actually needed, ref https://3v4l.org/umaZC

**Comment by samsonasik** [comment]
strrpos still possible, but with keep the strlen() usage, ref https://3v4l.org/I2N7t

File: src/View/Http/InjectTemplateListener.php [hunk 1]
```diff
@@ -172,8 +172,9 @@ protected function deriveControllerClass($controller)
             $controller = substr($controller, strrpos($controller, '\\') + 1);
         }
 
-        if ((10 < strlen($controller))
-            && ('Controller' == substr($controller, -10))
+        $length = strlen($controller);
+        if ($length > 10
+            && strrpos($controller, 'Controller') === ($length - 10)
```
Comment (michalbundyra) [inline src/View/Http/InjectTemplateListener.php]: Brackets are useless around `($length - 10)`.

**Review by michalbundyra — COMMENTED** [review]
Not convinced to this change, imo it decreased readability.

File: src/View/Http/InjectTemplateListener.php (hunk 1 above)
Comment (samsonasik) [reply src/View/Http/InjectTemplateListener.php]: done ;)

**Comment by Xerkus** [comment]
I would agree with @webimpress, this PR in its final form does not bring notable improvement. I am going to close this now.

**Pull request closed without merging**